src/sentry/static/sentry/app/components/discover/transactionsList.tsx

```diff
@@ -191,11 +208,67 @@ class TransactionsList extends React.Component<Props> {
     );
   }
 
+  renderTrendsTable(): React.ReactNode {
+    const {
+      trendView,
+      location,
+      selected,
+      organization,
+      cursorName,
+      linkDataTestId,
+      generateFirstLink,
+    } = this.props;
+    trendView.sorts = [selected.sort];
```

Comment:
Can this use `.withSorts` from `EventView`?

Reply:
Unfortunately no 😞, cause `trend_percentage()` isn't an actual field on the TrendView object https://github.com/getsentry/sentry/pull/22012/files#diff-4369fedead9ed7a463ffcb82ddff8ed5174981a69f0e3caf231fd6aee738de29R134
The validation in `withSorts` removes the sort instead.

Reply:
Oof, this is one of those columns inserted by the endpoint right? Doesn't have to be in this PR but perhaps consider having `TrendView` inheriting from `EventView` instead of just a new type based on it. That way these specialized behaviours can be handled in the `TrendView` class.

Reply:
:+1: Yep, I'll backlog a task to change how we do that